packages/zwave-js/src/lib/commandclass/DoorLockLoggingCC.ts

```diff
@@ -0,0 +1,662 @@
+import {
+	CommandClasses,
+	enumValuesToMetadataStates,
+	Maybe,
+	MessageOrCCLogEntry,
+	validatePayload,
+	ValueID,
+	ValueMetadata,
+	ZWaveError,
+	ZWaveErrorCodes,
+} from "@zwave-js/core";
+import { pick } from "@zwave-js/shared";
+import type { Driver } from "../driver/Driver";
+import { MessagePriority } from "../message/Constants";
+import {
+	PhysicalCCAPI,
+	PollValueImplementation,
+	POLL_VALUE,
+	throwUnsupportedProperty,
+} from "./API";
+import {
+	API,
+	CCCommand,
+	CCCommandOptions,
+	ccValue,
+	ccValueMetadata,
+	CommandClass,
+	commandClass,
+	CommandClassDeserializationOptions,
+	expectedCCResponse,
+	gotDeserializationOptions,
+	implementedVersion,
+} from "./CommandClass";
+
+interface DateSegments {
+	year: number;
+	month: number;
+	day: number;
+	hour: number;
+	minute: number;
+	second: number;
+}
+
+function segmentsToDate(segments: DateSegments): Date {
+	return new Date(
+		segments.year,
+		segments.month,
+		segments.day,
+		segments.hour,
+		segments.minute,
+		segments.second,
+	);
+}
+
+// All the supported commands
+export enum DoorLockLoggingCommand {
+	RecordsSupportedGet = 0x01,
+	RecordsSupportedReport = 0x02,
+	RecordGet = 0x03,
+	RecordReport = 0x04,
+}
+
+// @publicAPI
+export enum EventType {
+	LockCode = 0x01,
+	UnlockCode = 0x02,
+	LockButton = 0x03,
+	UnlockButton = 0x04,
+	LockCodeOOSchedule = 0x05,
+	UnlockCodeOOSchedule = 0x06,
+	IllegalCode = 0x07,
+	LockManual = 0x08,
+	UnlockManual = 0x09,
+	LockAuto = 0x0a,
+	UnlockAuto = 0x0b,
+	LockRemoteCode = 0x0c,
+	UnlockRemoteCode = 0x0d,
+	LockRemote = 0x0e,
+	UnlockRemote = 0x0f,
+	LockRemoteCodeOOSchedule = 0x10,
+	UnlockRemoteCodeOOSchedule = 0x11,
+	RemoteIllegalCode = 0x12,
+	LockManual2 = 0x13,
+	UnlockManual2 = 0x14,
+	LockSecured = 0x15,
+	LockUnsecured = 0x16,
+	UserCodeAdded = 0x17,
+	UserCodeDeleted = 0x18,
+	AllUserCodesDeleted = 0x19,
+	MasterCodeChanged = 0x1a,
+	UserCodeChanged = 0x1b,
+	LockReset = 0x1c,
+	ConfigurationChanged = 0x1d,
+	LowBattery = 0x1e,
+	NewBattery = 0x1f,
+	Unknown = 0x20,
+}
+
+const eventTypeLabel: { [key in keyof typeof EventType]: string } = {
+	LockCode: "Locked via Access Code",
+	UnlockCode: "Unlocked via Access Code",
+	LockButton: "Locked via Lock Button",
+	UnlockButton: "Unlocked via Unlock Button",
+	LockCodeOOSchedule: "Lock Attempt via Out of Schedule Access Code",
+	UnlockCodeOOSchedule: "Unlock Attempt via Out of Schedule Access Code",
```

Comment:
```suggestion
	LockCodeOOSchedule: "Out of Schedule Lock Attempt via Access Code",
	UnlockCodeOOSchedule: "Out of Schedule Unlock Attempt via Access Code",
```
I think this rolls off the tongue a bit better. Do the same for the other ones that are out of schedule